### ISSUE TITLE
Training issues

### ISSUE THREAD

**Issue by ZMH-SDUST** [gpt4vision/OvSGTR#16]
When I used the default scripts (.sh files) provided by the code to train the model, I could not reach the level reported in the paper under various experimental settings. However, the performance of the weight files provided by the author was reliable. Therefore, I wondered if there was something wrong with my training settings?

**Comment by ZMH-SDUST** [comment]
I ran a series of experiments with Swin_T as the backbone, and it seems that the code does not use knowledge distillation by default. Is this the reason?

**Comment by JosephChenHub** [comment]
> When I used the default scripts (.sh files) provided by the code to train the model, I could not reach the level reported in the paper under various experimental settings. However, the performance of the weight files provided by the author was reliable. Therefore, I wondered if there was something wrong with my training settings?

could you provide more details of your training ?

**Comment by ZMH-SDUST** [comment]
Thank you for your reply. Here is the training script I used: "Nohup bash scripts/DINO_train.sh vg ./config/GroundingDINO_SwinT_OGC_ovdr.py ./data ./logs/ovsgtr_vg_swint_ovdr ./GroundingDINO/weights/groundingdino_swint_ogc.pth > output.txt 2>&1 &"

I did not change the training parameters. The experimental device is RTX 3090Ti. This is the prediction result of the model：

SGG eval :     R @ 20: 0.0323;     R @ 50: 0.0480;     R @ 100: 0.0626;  for mode=sgdet, type=Recall(Main).
SGG eval :  ng-R @ 20: 0.0317;  ng-R @ 50: 0.0472;  ng-R @ 100: 0.0642;  for mode=sgdet, type=No Graph Constraint Recall(Main).
SGG eval :    zR @ 20: 0.0044;    zR @ 50: 0.0111;    zR @ 100: 0.0111;  for mode=sgdet, type=Zero Shot Recall.
SGG eval : ng-zR @ 20: 0.0000; ng-zR @ 50: 0.0133; ng-zR @ 100: 0.0156;  for mode=sgdet, type=No Graph Constraint Zero Shot Recall.
SGG eval :    zR @ 20: 0.0262;    zR @ 50: 0.0431;    zR @ 100: 0.0608;  for mode=sgdet, type=Ovd Zero Shot Recall.
SGG eval :    zR @ 20: 0.0000;    zR @ 50: 0.0000;    zR @ 100: 0.0000;  for mode=sgdet, type=Ovr Zero Shot Recall.
SGG eval :    mR @ 20: 0.0134;    mR @ 50: 0.0218;    mR @ 100: 0.0294;  for mode=sgdet, type=Mean Recall.

**Comment by ZMH-SDUST** [comment]
The script under ovr settings and the corresponding experimental results:

nohup bash scripts/DINO_train.sh vg ./config/GroundingDINO_SwinT_OGC_ovr.py  ./data  ./logs/ovsgtr_vg_swint_ovr ./GroundingDINO/weights/groundingdino_swint_ogc.pth > output.txt 2>&1 &

====================================================================================================
SGG eval :     R @ 20: 0.1049;     R @ 50: 0.1303;     R @ 100: 0.1465;  for mode=sgdet, type=Recall(Main).
SGG eval :  ng-R @ 20: 0.1064;  ng-R @ 50: 0.1459;  ng-R @ 100: 0.1746;  for mode=sgdet, type=No Graph Constraint Recall(Main).
SGG eval :    zR @ 20: 0.0000;    zR @ 50: 0.0156;    zR @ 100: 0.0200;  for mode=sgdet, type=Zero Shot Recall.
SGG eval : ng-zR @ 20: 0.0000; ng-zR @ 50: 0.0133; ng-zR @ 100: 0.0311;  for mode=sgdet, type=No Graph Constraint Zero Shot Recall.
SGG eval :    zR @ 20: 0.0001;    zR @ 50: 0.0004;    zR @ 100: 0.0005;  for mode=sgdet, type=Ovr Zero Shot Recall.
SGG eval :    mR @ 20: 0.0454;    mR @ 50: 0.0575;    mR @ 100: 0.0677;  for mode=sgdet, type=Mean Recall.

**Comment by JosephChenHub** [comment]
> The script under ovr settings and the corresponding experimental results:
> 
> nohup bash scripts/DINO_train.sh vg ./config/GroundingDINO_SwinT_OGC_ovr.py ./data ./logs/ovsgtr_vg_swint_ovr ./GroundingDINO/weights/groundingdino_swint_ogc.pth > output.txt 2>&1 &
> 
> ==================================================================================================== SGG eval : R @ 20: 0.1049; R @ 50: 0.1303; R @ 100: 0.1465; for mode=sgdet, type=Recall(Main). SGG eval : ng-R @ 20: 0.1064; ng-R @ 50: 0.1459; ng-R @ 100: 0.1746; for mode=sgdet, type=No Graph Constraint Recall(Main). SGG eval : zR @ 20: 0.0000; zR @ 50: 0.0156; zR @ 100: 0.0200; for mode=sgdet, type=Zero Shot Recall. SGG eval : ng-zR @ 20: 0.0000; ng-zR @ 50: 0.0133; ng-zR @ 100: 0.0311; for mode=sgdet, type=No Graph Constraint Zero Shot Recall. SGG eval : zR @ 20: 0.0001; zR @ 50: 0.0004; zR @ 100: 0.0005; for mode=sgdet, type=Ovr Zero Shot Recall. SGG eval : mR @ 20: 0.0454; mR @ 50: 0.0575; mR @ 100: 0.0677; for mode=sgdet, type=Mean Recall.

the pretrained checkpoint should use models after relation-aware pre-training like https://huggingface.co/JosephZ/OvSGTR/blob/main/vg-pretrain-coco-swint.pth

**Comment by ZMH-SDUST** [comment]
Thank you very much for your patient reply.

You mean, taking backbone as Swin_T, in close-set and ovd mode, use "./GroundingDINO/weights/groundingdino_swint_ogc.pth", but in ovr and ovdr mode, use vg-pretrain-coco-swint.pth as the pre-trained weight file?

In addition, in ovd mode, under Swin-T backbone, the weight file you provided shows 12.34/18.14/23.20 and 6.90/12.06/16.49 recall in "base+novel" and "Novel" modes. But the performance of the results we obtained from training is relatively low, namely 10.16/15.20/19.53 and 4.13/7.91/11.50. What do you think is the reason?

====================================================================================================
SGG eval :     R @ 20: 0.1016;     R @ 50: 0.1520;     R @ 100: 0.1953;  for mode=sgdet, type=Recall(Main).
SGG eval :  ng-R @ 20: 0.1042;  ng-R @ 50: 0.1544;  ng-R @ 100: 0.2025;  for mode=sgdet, type=No Graph Constraint Recall(Main).
SGG eval :    zR @ 20: 0.0178;    zR @ 50: 0.0222;    zR @ 100: 0.0311;  for mode=sgdet, type=Zero Shot Recall.
SGG eval : ng-zR @ 20: 0.0178; ng-zR @ 50: 0.0267; ng-zR @ 100: 0.0311;  for mode=sgdet, type=No Graph Constraint Zero Shot Recall.
SGG eval :    zR @ 20: 0.0413;    zR @ 50: 0.0791;    zR @ 100: 0.1150;  for mode=sgdet, type=Ovd Zero Shot Recall.
SGG eval :    mR @ 20: 0.0271;    mR @ 50: 0.0412;    mR @ 100: 0.0510;  for mode=sgdet, type=Mean Recall.

**Comment by JosephChenHub** [comment]
> Thank you very much for your patient reply.
> 
> You mean, taking backbone as Swin_T, in close-set and ovd mode, use "./GroundingDINO/weights/groundingdino_swint_ogc.pth", but in ovr and ovdr mode, use vg-pretrain-coco-swint.pth as the pre-trained weight file?
> 
> In addition, in ovd mode, under Swin-T backbone, the weight file you provided shows 12.34/18.14/23.20 and 6.90/12.06/16.49 recall in "base+novel" and "Novel" modes. But the performance of the results we obtained from training is relatively low, namely 10.16/15.20/19.53 and 4.13/7.91/11.50. What do you think is the reason?
> 
> ==================================================================================================== SGG eval : R @ 20: 0.1016; R @ 50: 0.1520; R @ 100: 0.1953; for mode=sgdet, type=Recall(Main). SGG eval : ng-R @ 20: 0.1042; ng-R @ 50: 0.1544; ng-R @ 100: 0.2025; for mode=sgdet, type=No Graph Constraint Recall(Main). SGG eval : zR @ 20: 0.0178; zR @ 50: 0.0222; zR @ 100: 0.0311; for mode=sgdet, type=Zero Shot Recall. SGG eval : ng-zR @ 20: 0.0178; ng-zR @ 50: 0.0267; ng-zR @ 100: 0.0311; for mode=sgdet, type=No Graph Constraint Zero Shot Recall. SGG eval : zR @ 20: 0.0413; zR @ 50: 0.0791; zR @ 100: 0.1150; for mode=sgdet, type=Ovd Zero Shot Recall. SGG eval : mR @ 20: 0.0271; mR @ 50: 0.0412; mR @ 100: 0.0510; for mode=sgdet, type=Mean Recall.
exactly. 
1. check the batch size (typically we use 4x or 8x GPUs); 2. the default evaluation is on the validation set instead of the test set during training.

**Comment by ZMH-SDUST** [comment]
Thank you for your patient response and look forward to your future work in the SGG field.

**Comment by langlangmingyue** [comment]
> Thank you for your reply. Here is the training script I used: "Nohup bash scripts/DINO_train.sh vg ./config/GroundingDINO_SwinT_OGC_ovdr.py ./data ./logs/ovsgtr_vg_swint_ovdr ./GroundingDINO/weights/groundingdino_swint_ogc.pth > output.txt 2>&1 &"
> 
> I did not change the training parameters. The experimental device is RTX 3090Ti. This is the prediction result of the model：
> 
> SGG eval : R @ 20: 0.0323; R @ 50: 0.0480; R @ 100: 0.0626; for mode=sgdet, type=Recall(Main). SGG eval : ng-R @ 20: 0.0317; ng-R @ 50: 0.0472; ng-R @ 100: 0.0642; for mode=sgdet, type=No Graph Constraint Recall(Main). SGG eval : zR @ 20: 0.0044; zR @ 50: 0.0111; zR @ 100: 0.0111; for mode=sgdet, type=Zero Shot Recall. SGG eval : ng-zR @ 20: 0.0000; ng-zR @ 50: 0.0133; ng-zR @ 100: 0.0156; for mode=sgdet, type=No Graph Constraint Zero Shot Recall. SGG eval : zR @ 20: 0.0262; zR @ 50: 0.0431; zR @ 100: 0.0608; for mode=sgdet, type=Ovd Zero Shot Recall. SGG eval : zR @ 20: 0.0000; zR @ 50: 0.0000; zR @ 100: 0.0000; for mode=sgdet, type=Ovr Zero Shot Recall. SGG eval : mR @ 20: 0.0134; mR @ 50: 0.0218; mR @ 100: 0.0294; for mode=sgdet, type=Mean Recall.

Hello, may I ask how many 3090s you used when reproducing this code and how long the training process took? I'm looking forward to your reply, which will be of great significance to me
